columnflow/calibration/cms/jets.py

```diff
@@ -111,58 +115,130 @@ def ak_evaluate(evaluator: correctionlib.highlevel.Correction, *args) -> float:
 #
 
 # define default functions for jec calibrator
-def get_jec_file_default(self, external_files: DotDict) -> str:
+def get_jerc_file_default(self, external_files: DotDict) -> str:
     """
-    Function to obtain external jec files.
+    Function to obtain external correction files for JEC and/or JER.
 
     By default, this function extracts the location of the jec correction
-    files from the current config instance *config_inst*:
+    files from the current config instance *config_inst*. The key of the
+    external file depends on the jet collection. For ``Jet`` (AK4 jets), this
+    resolves to ``jet_jerc``, and for ``FatJet`` it is resolved to
+    ``fatJet_jerc``.
 
     .. code-block:: python
 
         cfg.x.external_files = DotDict.wrap({
             "jet_jerc": "/afs/cern.ch/work/m/mrieger/public/mirrors/jsonpog-integration-9ea86c4c/POG/JME/2017_UL/jet_jerc.json.gz",
+            "fatJet_jerc": "/afs/cern.ch/work/m/mrieger/public/mirrors/jsonpog-integration-9ea86c4c/POG/JME/2017_UL/fatJet_jerc.json.gz",
         })
 
     :param external_files: Dictionary containing the information about the file location
     :return: path or url to correction file(s)
     """ # noqa
-    return external_files.jet_jerc
+
+    # get config
+    try_attrs = ("get_jec_config", "get_jer_config")
+    jerc_config = None
+    for try_attr in try_attrs:
+        try:
+            jerc_config = getattr(self, try_attr)()
+        except AttributeError:
+            continue
+        else:
+            break
+
+    # fail if not found
+    if jerc_config is None:
+        raise ValueError(
+            "could not retrieve jer/jec config, none of the following methods "
+            f"were found: {try_attrs}",
+        )
+
+    # first check config for user-supplied `external_file_key`
+    ext_file_key = jerc_config.get("external_file_key", None)
+    if ext_file_key is not None:
+        return external_files[ext_file_key]
+
+    # if not found, try to resolve from jet collection name and fail if not standard NanoAOD
+    if self.jet_name not in get_jerc_file_default.map_jet_name_file_key:
+        available_keys = ", ".join(sorted(get_jerc_file_default.map_jet_name_file_key))
+        raise ValueError(
+            f"could not determine external file key for jet collection '{self.jet_name}', "
+            f"name is not one of standard NanoAOD jet collections: {available_keys}",
+        )
+
+    # return external file
+    ext_file_key = get_jerc_file_default.map_jet_name_file_key[self.jet_name]
+    return external_files[ext_file_key]
+
+
+# default external file keys for known jet collections
+get_jerc_file_default.map_jet_name_file_key = {
+    "Jet": "jet_jerc",
+    "FatJet": "fatJet_jerc",
+}
 
 
 def get_jec_config_default(self) -> DotDict:
```

Comment:
Could you add a type hint `self: Calibrator` here? Just to reflect to which object this function will be bound.